### PR TITLE
upgrade to level 3 of psalm

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**
I don't like so many if conditions inside class `InputRequest`.
But I have found this method because otherwise psalm tests don't pass.

Any tips to improve the code?

### CODE REVIEW

File: src/UI/SingleCommandApplication.php [hunk 1]
```diff
@@ -37,10 +37,19 @@ class SingleCommandApplication extends Application
...
      */
     public function __construct(Command $command, string $version = 'UNKNOWN')
     {
-        parent::__construct($command->getName(), $version);
+        $commandName = $command->getName();
+        if (null === $commandName) {
+            throw new \InvalidArgumentException('command cannot be null');
+        }
+
+        $name =
```
Comment (JellyBellyDev) [inline src/UI/SingleCommandApplication.php:50]: what? :)

**Review by JellyBellyDev — CHANGES_REQUESTED** [review]

File: src/UI/SingleCommandApplication.php (hunk 1 above)
Comment (AlessandroMinoccheri) [reply src/UI/SingleCommandApplication.php]: Interesting... Fixed... but how could he pass the build earlier?

Comment (JellyBellyDev) [reply src/UI/SingleCommandApplication.php]: I asked myself the same question...

**Review by JellyBellyDev — APPROVED** [review]
LGTM

**Pull request merged**